mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1644,6 +1645,17 @@ def _log_eval_table(self):
         data = data.assign(**columns)
         artifact_file_name = f"{metric_prefix}{_EVAL_TABLE_FILE_NAME}"
         mlflow.log_table(data, artifact_file=artifact_file_name)
+        if self.eval_results_path:
+            from pyspark.sql import SparkSession
+
+            spark_session = SparkSession.getActiveSession()
+            if not spark_session:
+                spark_session = _create_local_spark_session_for_evaluate()
+            eval_table_spark = spark_session.createDataFrame(data)
+            eval_table_spark.write.mode("overwrite").format("delta").saveAsTable(
+                self.eval_results_path
+            )
```

Comment:
We should definitely automerge the schema. 

For append vs overwrite: 
* Append is the right thing to do if we add a new experiment to the same eval dataset.
* Overwrite is the right thing to do if the eval dataset has changed since last processed.
Can we differentiate between the two cases when the call is made?

Reply:
Makes sense! I think we should add an evaluator conf this. Our RAG solution can decide how to populate it

Reply:
Sounds good. Introduced a new evaluator conf `eval_results_mode` to control that.